travharv/execution_report.py

```diff
@@ -0,0 +1,212 @@
+        self.report_content["last_mod"] = last_mod
+
+
+class ExecutionReport:
+    """
+    A class to represent an ExecutionReport.
+    This class will report the results of the assertions per task per config.
+    e.g.: All TaskExecutionReports will be stored in a dictionary
+    """
+
+    def __init__(self, rdf_store_access: RDFStoreAccess, config_name: str):
+        """constructor
+
+        :param rdf_store_access: RDFStoreAccess
+        """
+        self.rdf_store_access = rdf_store_access
+        log.debug("ExecutionReport initialized")
+        self.report_content = {
+            "last_mod": timestamp(),
+            "config_name": config_name,
+            "id": uuid4(),
+        }
+        self.task_reports = []
+
+    def add_task_report(self, task_execution_report: TaskExecutionReport):
+        """
+        Report the results of the assertions to the store.
+        """
+        log.debug("Reporting the results of the execution report to the store")
+        # get info needed for update
+        task_id = task_execution_report.report_content["task_id"]
+        last_mod = task_execution_report.report_content["last_mod"]
+
+        log.debug(f"Task ID: {task_id}")
+        log.debug(f"Last Modified: {last_mod}")
+
+        # only if task execution report contains any assertion reports
+        # add it
+        # this is to circomvent the case where a task is executed
+        # but no assertions are made (e.g. no results came back
+        # from an initial SPARQL query)
+        if task_execution_report.assertion_reports.__len__() > 0:
+            self.task_reports.append(task_execution_report)
+        # to store with the report_content
+
+        log.debug(f"{task_execution_report.report_content['task_id']=}")
+        for assertion in task_execution_report.assertion_reports:
+            log.debug(f"{assertion.id=}")
+            log.debug(f"{assertion.assertion_path=}")
+            log.debug(f"{assertion.assertion_result=}")
+            log.debug(f"{assertion.message=}")
+            log.debug(f"Graphs added: {len(assertion.graph_reports)}")
+            # count up the ammount of triples that are
+            # located in assertion.graph_reports[i]["triples"]
+            c_triples = 0
+            for graph in assertion.graph_reports:
+                c_triples += graph.triple_count
+            log.debug(f"Triples added: {c_triples}")
+
+        self._make_report_graph()
+
+    def _make_report_graph(self):
+        pre_ttl = QUERY_BUILDER.build_syntax(
+            "execution_report.ttl",
+            execution_report=self.report_content,
+            task_reports=self.task_reports,
+        )
+
+        log.debug(f"{pre_ttl}")
+
+        self.execution_report_graph = Graph()
+        self.execution_report_graph.parse(data=pre_ttl, format="turtle")
+
+    def report_to_store(self):
+        self._make_report_graph()
+        self.rdf_store_access.insert_for_config(
```

Comment:
this seems the only point where rdf_store is needed
so it could be added in the argument-list of report_to_store and removed as an instance variable from the report class itself?